packages/ethereum-contracts/test/foundry/agreements/ConstantFlowAgreementV1.ACL.t.sol

```diff
@@ -155,17 +165,138 @@ contract ConstantFlowAgreementV1ACLTest is FoundrySuperfluidTester {
+
+        vm.expectRevert(IConstantFlowAgreementV1.CFA_ACL_NO_SENDER_FLOW_OPERATOR.selector);
+        vm.startPrank(flowOperator);
+        superToken.increaseFlowAllowanceWithPermissions(flowOperator, permissions, 1);
+        vm.stopPrank();
+    }
+
+    function testRevertIfDecreaseFlowAllowanceWithPermissionsToSelf(address flowOperator, uint8 permissions) public {
+        vm.assume(FlowOperatorDefinitions.isPermissionsClean(permissions));
+
+        vm.expectRevert(IConstantFlowAgreementV1.CFA_ACL_NO_SENDER_FLOW_OPERATOR.selector);
+        vm.startPrank(flowOperator);
+        superToken.decreaseFlowAllowanceWithPermissions(flowOperator, permissions, 0);
+        vm.stopPrank();
+    }
+
+    function testRevertIfIncreaseFlowAllowanceWithNegativeAllowance(
+        address flowOperator,
+        uint8 permissions,
+        int96 flowAllowance
+    ) public {
+        vm.assume(FlowOperatorDefinitions.isPermissionsClean(permissions));
+        vm.assume(flowAllowance < 0);
+        vm.assume(address(this) != flowOperator);
+
+        vm.expectRevert(IConstantFlowAgreementV1.CFA_ACL_NO_NEGATIVE_ALLOWANCE.selector);
+        superToken.increaseFlowAllowanceWithPermissions(flowOperator, permissions, flowAllowance);
+    }
+
+    function testRevertIfDecreaseFlowAllowanceWithNegativeAllowance(
+        address flowOperator,
+        uint8 permissions,
+        int96 flowAllowance
+    ) public {
+        vm.assume(FlowOperatorDefinitions.isPermissionsClean(permissions));
+        vm.assume(flowAllowance < 0);
+        vm.assume(address(this) != flowOperator);
+
+        vm.expectRevert(IConstantFlowAgreementV1.CFA_ACL_NO_NEGATIVE_ALLOWANCE.selector);
+        superToken.decreaseFlowAllowanceWithPermissions(flowOperator, permissions, flowAllowance);
+    }
+
+    function testRevertIfIncreaseFlowRateAllowanceWithPermissionsOverflows(address flowOperator, uint8 permissions)
+        public
+    {
+        vm.assume(FlowOperatorDefinitions.isPermissionsClean(permissions));
+        vm.assume(address(this) != flowOperator);
+
+        superToken.increaseFlowAllowanceWithPermissions(flowOperator, permissions, type(int96).max);
+        vm.expectRevert("CallUtils: target panicked: 0x11");
+        superToken.increaseFlowRateAllowance(flowOperator, 1);
+    }
+
+    function testIncreaseFlowAllowanceWithPermissions(address flowOperator, uint8 permissions, int96 flowAllowance)
+        public
+    {
+        vm.assume(FlowOperatorDefinitions.isPermissionsClean(permissions));
+        vm.assume(flowAllowance > 0);
+        vm.assume(address(this) != flowOperator);
+
+        (bytes32 flowOperatorId, uint8 oldPermissions, int96 oldFlowRateAllowance) =
+            sf.cfa.getFlowOperatorData(superToken, address(this), flowOperator);
+
+        superToken.increaseFlowAllowanceWithPermissions(flowOperator, permissions, flowAllowance);
+
+        uint8 newPermissions = sf.cfa.deltaAddPermissions(oldPermissions, permissions);
```

Comment:
I'm not sure I like this...

Feels like testing the implementation with _the implementation_.

Is `deltaAddPermissions` covered elsewhere?

Reply:
we are not explicitly testing `deltaAddPermissions`.